packages/react-aria-components/src/Autocomplete.tsx

```diff
@@ -58,7 +60,9 @@ export function UNSTABLE_Autocomplete(props: AutocompleteProps) {
     <Provider
       values={[
         [UNSTABLE_AutocompleteStateContext, state],
-        [InputContext, {...inputProps, ref: inputRef}],
+        [SearchFieldContext, textFieldProps],
+        [TextFieldContext, textFieldProps],
+        [InputContext, {ref: inputRef}],
```

Comment:
can't we pass the inputRef to SearchField or TextField Context now?

Reply:
if so can we remove the InputContext merging?

Reply:
if I pass the ref down through context, it will be attached to the wrapping div rather than the input itself so I still need the InputContext. If I can get rid of the checks against `inputRef` in useAutocomplete then I can get rid of it but digging for alternatives right now

Reply:
remind me why we need a document-level capturing listener?

Reply:
We need to dispatch a keyup event onto the ListBox link item so that we can properly simulate a Enter press and trigger a the Listbox's link navigation